app/lib/screens/wallets/send.dart

```diff
@@ -322,26 +343,80 @@ class _WalletSendScreenState extends ConsumerState<WalletSendScreen> {
                       ),
                     ),
                   if (chainType == ChainType.Stellar)
-                    ListTile(
-                      title: TextField(
-                          style: Theme.of(context)
-                              .textTheme
-                              .bodyMedium!
-                              .copyWith(
-                                color: Theme.of(context).colorScheme.onSurface,
+                    Padding(
+                      padding: const EdgeInsets.symmetric(
+                          vertical: 8, horizontal: 10),
+                      child: SizedBox(
+                        width: double.infinity,
+                        child: Row(
+                          children: [
+                            Expanded(
+                              child: TextField(
+                                style: Theme.of(context)
+                                    .textTheme
+                                    .bodyMedium!
+                                    .copyWith(
+                                      color: Theme.of(context)
+                                          .colorScheme
+                                          .onSurface,
+                                    ),
+                                controller: memoController,
+                                decoration: InputDecoration(
+                                  errorText: memoError,
+                                  labelText:
+                                      _memoTypeToString(selectedMemoType),
+                                ),                              
+                                onChanged: (value) {
+                                  setState(() {
+                                    memoError =
+                                        _validateMemo(value, selectedMemoType);
+                                  });
+                                },
+                              ),
+                            ),
+                            const SizedBox(width: 10),
+                            SizedBox(
+                              width: 150,
+                              child: DropdownButtonFormField<MemoType>(
+                                value: selectedMemoType,
+                                onChanged: (MemoType? newValue) {
+                                  setState(() {
+                                    selectedMemoType = newValue!;
+                                  });
+                                },
+                                items: MemoType.values.map((MemoType type) {
+                                  return DropdownMenuItem<MemoType>(
+                                    value: type,
+                                    child: Text(
+                                      _memoTypeToString(type),

```

Comment:
`Memo Hash` or `Memo Text` looks too long, maybe we can say `Hash` or `Text` only.